homeassistant/components/point/__init__.py

```diff
@@ -70,62 +78,96 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     conf = config[DOMAIN]
 
-    config_flow.register_flow_implementation(
-        hass, DOMAIN, conf[CONF_CLIENT_ID], conf[CONF_CLIENT_SECRET]
+    await async_import_client_credential(
+        hass,
+        DOMAIN,
+        ClientCredential(
+            conf[CONF_CLIENT_ID],
+            conf[CONF_CLIENT_SECRET],
+        ),
+    )
+
+    async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2025.4.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Point",
+        },
     )
 
     hass.async_create_task(
         hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
         )
     )
 
     return True
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Point from a config entry."""
+async def async_setup_entry(hass: HomeAssistant, entry: PointConfigEntry) -> bool:
+    """Set up Minut Point from a config entry."""
 
-    async def token_saver(token, **kwargs):
-        _LOGGER.debug("Saving updated token %s", token)
+    if "auth_implementation" not in entry.data:
+        # Config entry is imported from old implementation without native oauth2.
         hass.config_entries.async_update_entry(
-            entry, data={**entry.data, CONF_TOKEN: token}
+            entry,
+            data={
+                **entry.data,
+                "auth_implementation": DOMAIN,
+                CONF_TOKEN: {
+                    **entry.data[CONF_TOKEN],
+                    "expires_at": 0,
+                },
+                "imported": True,
+            },
         )
```

Comment:
Let's just do proper reauth

Reply:
raise ConfigEntryAuthFailed

Reply:
Stole a line from: https://github.com/home-assistant/core/blob/dd10a833dbe50ecc571f5c65bf93b67944615fdb/homeassistant/components/tuya/__init__.py#L51

Reply:
Perhaps is it not needed as we handle it here: https://github.com/home-assistant/core/blob/c518b40f9bfc945bf6be2d9cba4054a5ea94d8ad/homeassistant/components/point/__init__.py#L144

Reply:
Yes but we want to raise auth failed if we are missing crucial information like the full token in the entry